tools/cli/admin_task_list_commands.go

```diff
@@ -145,3 +145,52 @@ func printTaskListStatus(taskListStatus *types.TaskListStatus) error {
 	}}
 	return RenderTable(os.Stdout, table, RenderOptions{Color: true})
 }
+
+func AdminUpdateTaskListPartitionConfig(c *cli.Context) error {
+	adminClient, err := getDeps(c).ServerAdminClient(c)
+	if err != nil {
+		return err
+	}
+	domain, err := getRequiredOption(c, FlagDomain)
+	if err != nil {
+		return commoncli.Problem("Required flag not found: ", err)
+	}
+	taskList, err := getRequiredOption(c, FlagTaskList)
+	if err != nil {
+		return commoncli.Problem("Required flag not found: ", err)
+	}
+	var taskListType *types.TaskListType
+	if strings.ToLower(c.String(FlagTaskListType)) == "activity" {
+		taskListType = types.TaskListTypeActivity.Ptr()
+	} else if strings.ToLower(c.String(FlagTaskListType)) == "decision" {
+		taskListType = types.TaskListTypeDecision.Ptr()
+	} else {
+		return commoncli.Problem("Invalid task list type: valid types are [activity, decision]", nil)
```

Comment:
tasklist type flag description says it is optional but you are returning err here. Let's change the description